aspnetcore/signalr/streaming.md

```diff
@@ -110,33 +160,72 @@ Console.WriteLine("Streaming completed");
 
 ::: moniker-end
 
+::: moniker range=">= aspnetcore-3.0"
+
+### Client to server streaming
+
+To invoke a client-to-server streaming hub method from the .NET client create a `Channel` and pass the `ChannelReader` as an argument to `SendAsync`, `InvokeAsync`, or `StreamAsChannelAsync`, depending on the hub method being invoked.
+
+Whenever data is written to the `ChannelWriter` the hub method on the server will receive a new item with the data from the client.
+
+To end the stream, complete the channel with `channel.Writer.TryComplete()`.
+
+```csharp
+var channel = Channel.CreateBounded<string>(10);
+await connection.SendAsync("UploadStream", channel.Reader);
+await channel.Writer.WriteAsync("some data");
+await channel.Writer.WriteAsync("some more data");
+channel.Writer.TryComplete();
+```
+
+::: moniker-end
+
 ## JavaScript client
 
-JavaScript clients call streaming methods on hubs by using `connection.stream`. The `stream` method accepts two arguments:
+### Server to client streaming
+
+JavaScript clients call server-to-client streaming methods on hubs by using `connection.stream`. The `stream` method accepts two arguments:
 
 * The name of the hub method. In the following example, the hub method name is `Counter`.
 * Arguments defined in the hub method. In the following example, the arguments are: a count for the number of stream items to receive, and the delay between stream items.
 
 `connection.stream` returns an `IStreamResult` which contains a `subscribe` method. Pass an `IStreamSubscriber` to `subscribe` and set the `next`, `error`, and `complete` callbacks to get notifications from the `stream` invocation.
 
-[!code-javascript[Streaming javascript](streaming/sample/wwwroot/js/stream.js?range=19-36)]
-
 ::: moniker range="= aspnetcore-2.1"
 
+[!code-javascript[Streaming javascript](streaming/sample.netcoreapp2.1/wwwroot/js/stream.js?range=19-36)]
+
 To end the stream from the client, call the `dispose` method on the `ISubscription` that is returned from the `subscribe` method.
 
 ::: moniker-end
 
 ::: moniker range=">= aspnetcore-2.2"
 
+[!code-javascript[Streaming javascript](streaming/sample.netcoreapp2.2/wwwroot/js/stream.js?range=19-36)]
+
 To end the stream from the client, call the `dispose` method on the `ISubscription` that is returned from the `subscribe` method. Calling this method will cause the `CancellationToken` parameter of the Hub method (if you provided one) to be canceled.
 
 ::: moniker-end
 
 ::: moniker range=">= aspnetcore-3.0"
 
+### Client to server streaming
+
+JavaScript clients call client-to-server streaming methods on hubs by passing in a `Subject` as one of the arguments in `send`, `invoke`, or `stream`, depending on the hub method being invoked. The `Subject` needs to be a class that looks like a `Subject`. For example, in RxJS you can use the [Subject](https://rxjs-dev.firebaseapp.com/api/index/class/Subject) class from that library.
```

Comment:
You can have multiple streaming arguments and put the streaming arguments in any position (e.g. they don't have to be the first arguments), right?

Phrasing like "by passing in a `Subject` as one of the arguments" is making me doubt this.

In the three "Client to server streaming" subsections, we should make it clear that any number of streaming arguments in any position is supported. We should have a sample that takes streams as the 1st and 3rd arguments or something.